compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -541,4 +555,54 @@ object Interactive {
     }
   }
 
+  /**
+   * Is this tree using a renaming introduced by an import statement?
+   *
+   * @param tree The tree to inspect
+   * @return True, if this tree's name is different than its symbol's name, indicating that
+   *         it uses a renaming introduced by an import statement.
```

Comment:
Aren't there cases where a tree's name is different from the symbol's name even without renaming ? e.g. I think when you do:
```scala
class A { self =>
 val x = self
 val y = this
}
```
Then the symbol of `self` is the same as the symbol of `this`.

Reply:
Good catch. It looks like the definition of `self` has a symbol of its own, but references to `self` have the same symbol as `this` (that is, the underlying class).

I fixed the doc in `def isRenamed` to mention that and will open an issue about renaming self-types.